src/CSET/operators/ageofair.py

```diff
@@ -0,0 +1,378 @@
+        least 2 time points to compute back trajectory.
+    plev: int
+        The pressure level of which to compute the back trajectory on. The function will search to
+        see if this exists and if not, will raise an exception.
+    incW: bool
+        If incW is True, then the back trajectories will take into account vertical velocity fields
+        and thus a back trajectory takes a 3D path back. In this case, vertical wind fields are
+        smoothed in addition to the prior coarsening, to prevent instantaneous sampling of strong
+        updrafts/downdrafts that would advect a parcel to the bottom/top of atmosphere if the
+        temporal sampling of the data is coarse enough (e.g. hourly or 3 hourly). This option enables
+        the trajectory to capture large areas of rising/sinking air, in phenomena such as overturning
+        circulations. If incW if False, then compute back trajectories on a pressure level with no
+        vertical advection (essentially a laminar, stratified flow).
+    cyclic: book
+        If cyclic is True, then the code will assume no east/west boundary and if a back trajectory
+        reaches the boundary, it will emerge out of the other side. This option is useful for large
+        domains such as the K-SCALE tropical channel, where there are only north/south boundaries in
+        the domain.
+
+    Returns
+    -------
+    iris.cube.Cube
+        An iris cube of the age of air data, with 3 dimensions (time, latitude, longitude).
+
+    """
+    # If not None, then use multiple cores to run age of air diagnostic.
+    multicore = None
+
+    # Set up temporary directory to store intermediate age of air slices.
+    tmpdir = tempfile.mkdtemp()
+    logging.info("Made tmpdir", tmpdir)
+
+    # Check that all cubes are of same size (will catch different dimension orders too).
+    if not XWIND.shape == YWIND.shape == WWIND.shape == GEOPOT.shape:
+        raise ValueError("Cubes are not the same shape")
+
+    # Get time units and assign for later
+    if str(XWIND.coord("time").units) == "hours since 1970-01-01 00:00:00":
+        timeunit = "hour"
+    else:
+        raise NotImplementedError("Unsupported time base")
+
+    # Make data non-lazy to speed up code.
+    logging.info("Making data non-lazy...")
+    x_arr = XWIND.data
+    y_arr = YWIND.data
+    z_arr = WWIND.data
+    g_arr = GEOPOT.data
+
+    # Smooth vertical velocity if using, to 2sigma (standard for 0.5 degree).
+    if incW:
+        logging.info("Smoothing vertical velocity...")
+        for t in range(0, z_arr.shape[0]):
+            for p in range(0, z_arr.shape[1]):
+                z_arr[t, p, :, :] = gaussian_filter(
+                    z_arr[t, p, :, :], [2, 2], mode="nearest"
+                )
+    else:
+        # If not using vertical velocity, set vertical velocity to zero.
+        z_arr[:] = 0
+
+    # Get time spacing of cube -
+    dt = XWIND.coord("time").points[1:] - XWIND.coord("time").points[:-1]
```

Comment:
What is this line meant to be doing? Is it getting the time corrected to have the first point at 0? Or the interval between all of the points?

Reply:
It computes the time difference between all points in the file to check they are the same (some model runs might produce output which is hourly up to say 48h, and then only output every 3 hours after this). Its to catch these cases. In theory we could compute the back trajectory on non monotonic time data, but it might complicate the interpretation (if the intervals go from hourly to 6 hourly, the back trajectory will be much less accurate for example), so for now not supporting this. Have updated comment to clarify this.